DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/src/mmhost.cpp

```diff
@@ -0,0 +1,149 @@
+#include <iomanip>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/interfaces.hpp>
+#include <sycl/ext/oneapi/annotated_arg/annotated_ptr.hpp>
+#include <sycl/sycl.hpp>
+#include "exception_handler.hpp"
+
+using namespace sycl;
+using namespace ext::oneapi::experimental;
+using usm_buffer_location =
+    ext::intel::experimental::property::usm::buffer_location;
+
+constexpr int kBL1 = 1;
+constexpr int kBL2 = 2;
+constexpr int kBL3 = 3;
+
+struct PointerIP {
+  int *const a;
+  int *const b;
+  int *const c;
+  int n;
+
+  void operator()() const {
+    for (int i = 0; i < n; i++) {
+      c[i] = a[i] + b[i];
+    }
+  }
+};
+
+struct MMHostIP {
+  annotated_ptr<int, decltype(properties{buffer_location<kBL1>, awidth<32>,
+                                         dwidth<32>, latency<0>,
+                                         read_write_mode_readwrite, maxburst<4>,
+                                         wait_request_requested})>
```

Comment:
This is shorter, let's use this. I wonder if Paul would prefer this too.
```suggestion
                                         wait_request<1>})>
```